### ISSUE TITLE
Define and document general conventions for commit messages, branches, filenames etc. 

### ISSUE THREAD

**Issue by sypets** [TYPO3-Documentation/T3DocTeam#19]
# To be defined conventions

* [ ] Naming of filenames and subdirectories for included files
* [ ]  Naming of filenames and subdirectories for images (or other assets)
* [ ] Naming of section anchors  (`.. _how-to-document:`) and intersphinx references, e.g. (`:ref:`how-to-document`) (see below)
* [ ]  Naming of branches, see also #39

In general, the conventions are documented in ["Writing documentation"](https://docs.typo3.org/typo3cms/HowToDocument)

# Already defined conventions

* [x] [Included files](https://docs.typo3.org/typo3cms/HowToDocument/Incoming/FileStructure.html#including-files) should be named with the ending  `.rst.txt`
* [x] [Usage of underlines for headers](https://docs.typo3.org/typo3cms/HowToDocument/WritingReST/HeadlinesAndSection.html)
* [x]  Commit message rules, see https://docs.typo3.org/typo3cms/HowToDocument/GeneralConventions/CommitMessages.html

# More details

* see https://docs.typo3.org/typo3cms/HowToDocument/GeneralConventions/CommitMessages.html

**Comment by DanielSiepmann** [comment]
Should we keep the commit message rules from TYPO3 Core?

For branches I would recommend `GitFlow` like `hotfix` and `feature` prefixes, followed by `/` and the issue number, if any, followed by a meaningful name.

**Comment by sypets** [comment]
@DanielSiepmann Yes, I think it is a good idea to use the same (or slightly modified) [commit message rules of the TYPO3 core](https://docs.typo3.org/typo3cms/ContributionWorkflowGuide/Appendix/GeneralTopics/CommitMessage.html). Advantage: 

* It will make it easier for core developers to switch back and forth. 
* You can use the same commit message template
* It will make browsing throught the commit log easier to read. 

We should change them slight, though:
* it does not make sense to use `[DOCS]`. We can however use 
  * `[BUGFIX]` for fixing things
  * `[FEATURE]` for new stuff
  * `[TASK]` for things such as cleanup tasks, such as this one

Example: 

```
[TASK] Cleanup Settings.cfg

Most importantly, describe what is changed with the commit
and not what is has not been working (that is part of the bug report already).

Resolves: #0
Related: #19
Releases: latest, TYPO3_8

```

* In github, using the issue number also works and a link to the issue is created
* We can also use Releases, so the person merging can backport (if we really want to do that)


Also, **I would not impose them as strictly yet**. I would use the "Rule" as Best practices / Recommendations. That way the more active contributors can use them and new people can just write simple commit messages and still get their patches merged without us having to constantly adapt the commit messages. 

This is a very pragmatic approach, I think, that is suited to the current workflow and nature of TYPO3 documentation.
